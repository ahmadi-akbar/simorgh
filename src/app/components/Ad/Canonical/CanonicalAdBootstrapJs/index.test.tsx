import React from 'react';
import { Helmet } from 'react-helmet';
import {
  render,
  act,
  waitFor,
} from '../../../react-testing-library-with-providers';
import CanonicalAdBootstrapJs from '.';

describe('CanonicalAds Ads', () => {
  describe('Snapshots', () => {
    it('should push dotcom bootstrap and configuration to a head script', () => {
      render(<CanonicalAdBootstrapJs />);

      expect(Helmet.peek().scriptTags[0]).toMatchSnapshot();
    });
  });

  describe('Assertions', () => {
    it('should set window.dotcomConfig with pageAds enabled', async () => {
      await act(async () => {
        render(<CanonicalAdBootstrapJs />);
      });

      await waitFor(() => {
        // @ts-expect-error dotcomConfig is added to the window object by BBC Ads script
        expect(window.dotcomConfig).toEqual({
          pageAds: true,
          playerAds: false,
        });
      });
    });

    it('should set window.dotcomConfig with adCampaign value provided', async () => {
      await act(async () => {
        render(<CanonicalAdBootstrapJs adcampaign="a-campaign" />);
      });

      await waitFor(() => {
        // @ts-expect-error dotcomConfig is added to the window object by BBC Ads script
        expect(window.dotcomConfig).toEqual({
          adcampaign: 'a-campaign',
          pageAds: true,
          playerAds: false,
        });
      });
    });
  });
});
