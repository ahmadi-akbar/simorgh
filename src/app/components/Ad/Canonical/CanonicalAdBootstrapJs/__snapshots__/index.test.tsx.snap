// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanonicalAds Ads Snapshots should push dotcom bootstrap and configuration to a head script 1`] = `
Object {
  "innerHTML": "
    window.dotcom = window.dotcom || { cmd: [] };
    window.dotcomConfig = {
      pageAds: true,
      playerAds: false
    };
    ",
  "type": "text/javascript",
}
`;
